### PR TITLE
Add property `type` to SlashCommandBuilder

### PR DESCRIPTION
**Pull request by immjs**
**Please describe the changes this PR makes and why it should be merged:**
Might help if people want to categorize their commands, instead of checking if the builder is a SlashCommandBuilder, they can simply check the `type` property of builder, very much like how `ContextMenuCommandBuilder` does it

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

**Comment by Qjuh** [comment]
This is not consistent builders design. You use the builder while you want to add to it/change it. When you‘re done you can `.toJSON()` it and will have an object having that type set already.

**Comment by immjs** [comment]
I had spent a few minutes looking at the toJSON and there is nothing adding `type` automagically. Simply, it is not required by the API.

If you want to talk about design, we can also talk about how ContextMenuCommandBuilder does have a `type` property yet SlashCommandBuilder doesn't, even though they end up in the same API call.

File: packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts [hunk 1]
```diff
@@ -65,6 +65,11 @@ export class SlashCommandBuilder {
 	 */
 	public readonly dm_permission: boolean | undefined = undefined;
 
+	/**
+	 * The type of this command
+	 */
+	public readonly type = 1;
```
Comment (jaw0r3k) [inline packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts:71]: Also you shouldn't use magic numbers, as they aren't readable

Comment (immjs) [reply packages/builders/src/interactions/slashCommands/SlashCommandBuilder.ts]: Alright, added the relevant type

**Comment by monbrey** [comment]
Sorry, I have to disagree. `ContextMenuCommandBuilder` has a type property because there are two types that have no other differences between them, and it's required by the API to define an application command as those types.

For the use-case you described developers can simply add a `type` property, `category` property, or literally any other property they like alongside the builder in the exports.

**Comment by immjs** [comment]
> Sorry, I have to disagree. `ContextMenuCommandBuilder` has a type property because there are two types that have no other differences between them, and it's required by the API to define an application command as those types.
> 
> For the use-case you described developers can simply add a `type` property, `category` property, or literally any other property they like alongside the builder in the exports.

Well, what are the downsides?

**Comment by jaw0r3k** [comment]
> Well, what are the downsides?

There is just no need to add it

**Comment by kyranet** [comment]
The `ContextMenuCommandBuilder` class has a `type` property because it uses the old builders design, which we will phase out on builders v2 based on the [RFC](https://github.com/discordjs/discord.js/issues/8015).

**Pull request closed without merging**